### DISCUSSION TITLE
Schedule chaining

### DISCUSSION THREAD

**Discussion by msiep** [borgbase/vorta#1211] [Ideas]
I'd find it very helpful if the Schedule tab had an option to start a backup after a specific other backup finished. Every night I have several backups I want to run. At the moment I have to estimate how long they'll take and leave enough time between the scheduled start times that there won't usually still be a backup running when another backup is scheduled to start. It would be excellent if I could just set a scheduled time for one of the backups, e.g. midnight, and then just set the order in which the other backups will run afterwards, e.g. backup A starts at midnight, backup B starts when backup A finishes, backup C starts with backup B finishes, etc.

**Comment by real-yfprojects** [comment]
Would an option like `Run only one backup at a time` solve your issue?

**Reply by msiep** [reply]
If it were worded "Run only one backup at a time" I wouldn't know whether that meant it would only run my scheduled backup if no other backup was already running at the scheduled time, and otherwise abort it, or would wait to start the backup once no other backup was running.

What would work would be an option after the scheduled start time like "If another backup is running at this time: Start this backup as soon as possible / Skip this backup"

The " as soon as possible" option would put it in a queue. So I could then achieve what I'm aiming for by scheduling my backups at, for example, 00:00, 00:01, 00:02, 00:03, 00:04, etc. - and they'd then actually just start at 00:00 and run in the same order as my start times, e.g. 00:00, 01:23, 02:14, 02:37, 05:02, 05:55.

**Reply by real-yfprojects** [reply]
> The " as soon as possible" option would put it in a queue. So I could then achieve what I'm aiming for by scheduling my backups at, for example, 00:00, 00:01, 00:02, 00:03, 00:04, etc. - and they'd then actually just start at 00:00 and run in the same order as my start times, e.g. 00:00, 01:23, 02:14, 02:37, 05:02, 05:55.

This is what I had in mind. Would you understand `Queue simultaneous backups`?

**Reply by msiep** [reply]
That would be better, but I think "Backup as soon as possible if another backup is in progress" would be more straightforward:
![20220303113814](https://user-images.githubusercontent.com/11274767/156630954-774e29fe-de43-48b0-bbc4-d747f0c15d62.png)
Some advantages of that over "Queue simultaneous backups" are:

- "as soon as possible" indirectly communicates that simultaneous backups are not possible (which I've confirmed from inspecting my Vorta logs).
- It's clearer that if you don't check the box, the backup will not happen if another backup is in progress at the scheduled time.
- It avoids possible misinterpretation of "simultaneous backups" as meaning "backups scheduled to start at exactly the same time".

**Reply by real-yfprojects** [reply]
> That would be better, but I think "Backup as soon as possible if another backup is in progress" would be more straightforward

A longer, more wordy phrase is always less ambiguous. But as your screenshot proves, it needs way more space. This should not be underestimated since text can expand up to 50% in some languages.

> - "as soon as possible" indirectly communicates that simultaneous backups are not possible (which I've confirmed from inspecting my Vorta logs).
They are if they go to different repositories.

> - It's clearer that if you don't check the box, the backup will not happen if another backup is in progress at the scheduled time.

This should not be the case if the box is unchecked. Recently vorta purposely got the feature to run multiple backups at the same time. I therefore don't think any of the current contributors has the intention to remove it again.

 
> - It avoids possible misinterpretation of "simultaneous backups" as meaning "backups scheduled to start at exactly the same time".

In German I would say "sich zeitlich ueberschneidende" backups but in English I am not that eloquent. However my translator translates that to _backups overlapping in time_.

**Comment by msiep** [comment]
> Recently vorta purposely got the feature to run multiple backups at the same time

Great. I'm on version 0.7.5 but look forward to that. In that case there's not as much need for what I was requesting, since the situation of scheduled backups being aborted won't happen anymore. But I'd guess it would still be good to have the option to run only one at a time. What do you think about this checkbox label, as a global setting on the Misc tab?:

[X] Allow simultaneous backups

It doesn't specify what happens if you uncheck it, but that can be specified in the documentation (and in a tooltip if Vorta were to start having tooltips). Ideally what would happen is queuing them, rather than aborting them as happens in version 0.7.5.

**Comment by m3nu** [comment]
Shouldn’t the new task queue either schedule backups one after another or run only one per repo at the same time? If this works as described, no new checkbox should be needed.

Managing dependencies between tasks even further is not something many users would benefit from, so I don't see this added.

**Reply by msiep** [reply]
I'm going to look into why I've still got version 0.7.5 rather than 0.8.3 despite having installed through Flatpak and being on Lubuntu 21.10. It sounds like if I had 0.8.3 I probably would not have submitted this idea.
